src/bp/core/services/dialog/flow/service.ts

```diff
@@ -126,12 +127,25 @@ export class FlowService {
     this.saveAll(botId, [flow])
   }
 
-  private prepareSaveFlow(flow) {
+  private async _prepareSaveFlow(botId: string, flow: FlowView): Promise<any> {
     const schemaError = validateFlowSchema(flow)
     if (schemaError) {
       throw new Error(schemaError)
     }
 
+    // Find and replace any node name that has changed though all the files that referenced the old name.
+    // FIXME Refactor: Use the node ID instead of the node name to reference it.
+    if (flow.location) {
```

Comment:
I have trouble to understand what's up in there... can you extract this in a function with a meaningful name. even if you plan to refactor this later.